icu4c/source/i18n/number_longnames.cpp

```diff
@@ -232,15 +282,96 @@ UnicodeString getPerUnitFormat(const Locale& locale, const UNumberUnitWidth &wid
     return UnicodeString(ptr, len);
 }
 
+// Load data originating from a deriveComponent CLDR element. The feature and
+// structure parameters must be null-terminated.
+void getDeriveComponentRule(Locale locale,
+                            const char *feature,
+                            const char *structure,
+                            UnicodeString &value0,
+                            UnicodeString &value1,
+                            UErrorCode &status) {
+    StackUResourceBundle derivationsBundle, stackBundle;
```

Comment:
I really like to separate the loading data methods from the class, we can implement those functions part of `UnitsData` or maybe `InflectionUnitsData`.

Also, it can be in a different PR or in the same PR

but having loading data in the same class make the class more complicated.

Reply:
Refactoring 3 year old code is out of scope for this PR, so I'm leaving this as is.

You're commenting on a function here, "void getDeriveComponentRule(...)" - so I don't know what class you're referring to as becoming "more complicated". And LongNameHandler itself exists precisely for the sake of loading and using units data.